test/t_fmemopen.c

```diff
@@ -30,26 +30,35 @@
 #if defined(__NetBSD__)
 #include <atf-c.h>
 #else
-#if defined(__linux__)
+#if defined(__PICOLIBC__)
+#define _GNU_SOURCE  /* strnlen */
+#elif defined(__linux__)
 #define _GNU_SOURCE
 #include <features.h>
```

Comment:
Somewhat unrelated comment: I initially just added a __PICOLIBC__ case as follows:
```
#if defined(__linux__)
#define _GNU_SOURCE
#include <features.h>
#endif
#if defined(__PICOLIBC__)
#define _GNU_SOURCE  /* strnlen */
#endif 
```

Interestingly that broke the tests since including features.h defines __GLIBC__ and then the wrong branches are compiled. I wonder if picolibc should install an empty features.h to avoid this subtle bug?

Reply:
Hrm. We've got sys/features.h which appears to do something similar. Maybe a top-level features.h that includes sys/features.h (or picolibc.h?)

Reply:
Actually, maybe the real solution is to remove system includes from the search path using --sysroot? But anyway this is unrelated and should IMO not block this PR.

Reply:
sorry, been afk for the weekend. thanks much for this fix.